bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/protocols/ScalarWebSystemProtocol.java

```diff
@@ -0,0 +1,1119 @@
+        if (StringUtils.isEmpty(cmd)) {
+            return;
+        }
+
+        final String localIrccUrl = irccUrl;
+        if (localIrccUrl == null || StringUtils.isEmpty(localIrccUrl)) {
+            logger.debug("IRCC URL was not specified in configuration");
+        } else {
+            try {
+                final IrccClient irccClient = new IrccClientFactory().get(localIrccUrl);
+                final ScalarWebContext context = getContext();
+                String localCmd = cmd;
+
+                final String cmdMap = context.getConfig().getCommandsMapFile();
+
+                final TransformationService localTransformService = context.getTransformService();
+                if (localTransformService != null && cmdMap != null) {
+                    String code;
+                    try {
+                        code = localTransformService.transform(cmdMap, cmd);
+
+                        if (StringUtils.isNotBlank(code)) {
+                            logger.debug("Transformed {} with map file '{}' to {}", cmd, cmdMap, code);
+
+                            try {
+                                localCmd = URLDecoder.decode(code, "UTF-8");
+                            } catch (final UnsupportedEncodingException e) {
+                                localCmd = code;
+                            }
+                        }
+                    } catch (final TransformationException e) {
+                        logger.error("Failed to transform {} using map file '{}', exception={}", cmd, cmdMap,
+                                e.getMessage());
+                        return;
+                    }
+                }
+
+                if (localCmd == null || StringUtils.isEmpty(localCmd)) {
+                    logger.debug("IRCC command was empty or null - ignoring");
+                    return;
+                }
+
+                // Always use an http transport to execute soap
+                HttpResponse httpResponse;
+                try (final SonyHttpTransport httpTransport = SonyTransportFactory
+                        .createHttpTransport(irccClient.getBaseUrl().toExternalForm())) {
+                    // copy all the options from the parent one (authentication options)
+                    getService().getTransport().getOptions().stream().forEach(o -> httpTransport.setOption(o));
+                    httpResponse = irccClient.executeSoap(httpTransport, localCmd);
+                } catch (final URISyntaxException e) {
+                    logger.debug("URI syntax exception: {}", e.getMessage());
+                    return;
+                }
+
+                switch (httpResponse.getHttpCode()) {
+                    case HttpStatus.OK_200:
+                        // everything is great!
+                        break;
+
+                    case HttpStatus.SERVICE_UNAVAILABLE_503:
+                        logger.debug("IRCC service is unavailable (power off?)");
```

Comment:
shouldn't this still warrant a change of status on the callback?

Reply:
No - individual services will stop responding depending on the state of the TV and all other services will operate perfectly fine.  In this case, the IRCC service stops responding if the TV is off but I don't want to take the thing OFFLINE because the system service is still operating (and thus you can use it to power the TV back on).